### PR TITLE
Merge release-boba into master

### PR DESCRIPTION
**Pull request by fabian-guerra**
This is the list of files that I had to manually merge:
```
- circle.yml
- include/mbgl/style/expression/let.hpp
- platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/geometry/LatLngBounds.java
- platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/MapGestureDetector.java
- platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/MapKeyListener.java
- platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/MapView.java
- platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/MapboxMap.java
- platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/Projection.java
- platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/Transform.java
- platform/android/MapboxGLAndroidSDK/src/test/java/com/mapbox/mapboxsdk/geometry/LatLngBoundsTest.java
- platform/android/MapboxGLAndroidSDKTestApp/src/main/AndroidManifest.xml
- platform/android/gradle/dependencies.gradle
- platform/android/src/example_custom_layer.cpp
- platform/android/src/geojson/point.cpp
- platform/darwin/src/NSPredicate+MGLAdditions.mm
- platform/darwin/test/MGLExpressionTests.mm
- platform/ios/Mapbox-iOS-SDK-nightly-dynamic.podspec
- platform/ios/Mapbox-iOS-SDK-symbols.podspec
- platform/ios/Mapbox-iOS-SDK.podspec
- platform/ios/app/MBXViewController.m
- platform/darwin/src/MGLTileSource.h
- src/mbgl/renderer/layers/render_custom_layer.cpp
- src/mbgl/style/conversion/filter.cpp
- src/mbgl/style/expression/interpolate.cpp
- src/mbgl/style/expression/value.cpp
- test/style/filter.test.cpp
```

### CODE REVIEW

File: circle.yml [hunk 1]
```diff
@@ -409,7 +422,7 @@ jobs:
       - deploy:
           name: Publish to Maven
           command: |
-            if [ "${CIRCLE_BRANCH}" == "master" ]; then make run-android-upload-archives ; fi
+            if [ "${CIRCLE_BRANCH}" == "release-boba" ]; then make run-android-upload-archives ; fi
```
Comment (tobrun) [inline circle.yml]: this change should be undone

File: platform/ios/CHANGELOG.md [hunk 2]
```diff
@@ -2,36 +2,91 @@
...
+* Labels are now transliterated from more languages when VoiceOver is enabled. ([#10881](https://github.com/mapbox/mapbox-gl-native/pull/10881))
 * Long-pressing the attribution button causes the SDK’s version number to be displayed in the action sheet that appears. ([#10650](https://github.com/mapbox/mapbox-gl-native/pull/10650))
+* Reduced offline download sizes for styles with symbol layers that render only icons, and no text. ([#11055](https://github.com/mapbox/mapbox-gl-native/pull/11055))
+* Added haptic feedback that occurs when the user rotates the map to due north, configurable via `MGLMapView.hapticFeedbackEnabled`. ([#10847](https://github.com/mapbox/mapbox-gl-native/pull/10847))
+* Added `MGLMapView.showsScale` as the recommended way to show the scale bar. This property can be set directly in Interface Builder. ([#11335](https://github.com/mapbox/mapbox-gl-native/pull/11335))
+* Fixed an issue where the scale bar would not appear until the map had moved. ([#11335](https://github.com/mapbox/mapbox-gl-native/pull/11335))
+
+## 3.7.6 - March 12, 2018
```
Comment (1ec5) [inline platform/ios/CHANGELOG.md]: The v3.7.6 and v3.7.5 sections duplicate existing content below.

File: platform/android/MapboxGLAndroidSDK/gradle.properties [hunk 3]
```diff
@@ -1,5 +1,5 @@
 GROUP=com.mapbox.mapboxsdk
-VERSION_NAME=7.0.0-SNAPSHOT
+VERSION_NAME=6.1.0-SNAPSHOT
```
Comment (tobrun) [inline platform/android/MapboxGLAndroidSDK/gradle.properties]: this change should be undone

**Comment by friedbunny** [comment]
I’ve un-required the `ios-sanitize-thread` CI build, as that was replaced by `ios-sanitize`. After this merges, I’ll update `master` settings here to require `ios-sanitize`, `ios-sanitize-address`, and `ios-static-analyzer`.

**Pull request merged**